site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -965,7 +995,7 @@ public function showDetails(string $gradeable_id) {
             foreach ($all_teams as $team) {
                 $student_ids = array_diff($student_ids, $team->getMembers());
                 $team_section = $gradeable->isGradeByRegistration() ? $team->getRegistrationSection() : $team->getRotatingSection();
-                if ($team->getSize() > 0 && (in_array($team_section, $sections) || $show_all)) {
+                if ($team->getSize() > 0 && (in_array($team_section, $sections, true) || $show_all)) {
```

Comment:
The cause for php-lint error I believe is that $team_section is of type int and $sections is of type array<string>, this is a type mismatch as either $sections should be type array<int> or $team_section should be type string.

Because of mismatching datatypes, this function is always returning false.

Reply:
So, I have used debugger and found out the following : 
team_section was actually giving string and
sections were giving array 

which are as intended, I still don't know the reason why I'm failing here